Cubical/Data/Vec/NAry.agda

```diff
@@ -0,0 +1,50 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Data.Vec.NAry where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+
+open import Cubical.Data.Nat
+open import Cubical.Data.Vec.Base
+
+private
+  variable
+    ℓ ℓ' : Level
+    A : Type ℓ
+    B : Type ℓ'
+
+nAryLevel : Level → Level → ℕ → Level
+nAryLevel ℓ₁ ℓ₂ zero    = ℓ₂
+nAryLevel ℓ₁ ℓ₂ (suc n) = ℓ-max ℓ₁ (nAryLevel ℓ₁ ℓ₂ n)
+
+nAryOp : (n : ℕ) → Type ℓ → Type ℓ' → Type (nAryLevel ℓ ℓ' n)
+nAryOp zero A B    = B
+nAryOp (suc n) A B = A → nAryOp n A B
+
+
+_$ⁿ_ : ∀ {n} → nAryOp n A B → (Vec A n → B)
+f $ⁿ []       = f
+f $ⁿ (x ∷ xs) = f x $ⁿ xs
+
+curryⁿ : ∀ {n} → (Vec A n → B) → nAryOp n A B
+curryⁿ {n = zero}  f   = f []
+curryⁿ {n = suc n} f x = curryⁿ (λ xs → f (x ∷ xs))
+
+$ⁿ-curryⁿ : ∀ {n} (f : Vec A n → B) → _$ⁿ_ (curryⁿ f) ≡ f
+$ⁿ-curryⁿ {n = zero} f  = funExt λ { [] → refl }
+$ⁿ-curryⁿ {n = suc n} f = funExt λ { (x ∷ xs) i → $ⁿ-curryⁿ {n = n} (λ ys → f (x ∷ ys)) i xs}
+
+curryⁿ-$ⁿ : ∀ {n} (f : nAryOp {ℓ = ℓ} {ℓ' = ℓ'} n A B) → curryⁿ {A = A} {B = B} (_$ⁿ_ f) ≡ f
+curryⁿ-$ⁿ {n = zero} f  = refl
+curryⁿ-$ⁿ {n = suc n} f = funExt λ x → curryⁿ-$ⁿ {n = n} (f x)
+
+nAryOp≃VecFun : ∀ {n} → nAryOp n A B ≃ (Vec A n → B)
```

Comment:
@Saizan : I couldn't figure out how to apply `ua` to this to get a Path... When trying I got all kinds of yellow that I didn't manage to get rid of

Reply:
I guess the problem is that agda only sees those two types have the same universe level for concrete values of `n`.

Maybe the easiest would be for `nAryOp` to return a type in the max of the two levels and use `Lift` in the `zero` case? Or some other variation that doesn't make the level depend on the `n`.

Reply:
That might indeed work! But it was so annoying to add Lift everywhere... I have no uses of this equality yet so I'll keep things as they are but with a comment